browser/brave_rewards/test/creator_detection_browsertest.cc

```diff
@@ -0,0 +1,349 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include "base/memory/raw_ref.h"
+#include "base/test/bind.h"
+#include "base/test/scoped_feature_list.h"
+#include "base/test/test_future.h"
+#include "base/test/test_timeouts.h"
+#include "base/timer/timer.h"
+#include "brave/browser/brave_rewards/rewards_tab_helper.h"
+#include "brave/components/brave_rewards/common/features.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "net/dns/mock_host_resolver.h"
+
+using net::test_server::BasicHttpResponse;
+using net::test_server::HttpRequest;
+using net::test_server::HttpResponse;
+
+namespace brave_rewards {
+
+class CreatorDetectionBrowserTest : public InProcessBrowserTest {
+ protected:
+  CreatorDetectionBrowserTest() {
+    scoped_feature_list_.InitWithFeatures(
+        {features::kPlatformCreatorDetectionFeature}, {});
+  }
+
+  ~CreatorDetectionBrowserTest() override = default;
+
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    host_resolver()->AddRule("*", "127.0.0.1");
+    embedded_https_test_server().SetCertHostnames(
+        {"twitter.com", "github.com", "api.github.com", "reddit.com",
+         "www.twitch.tv", "vimeo.com", "www.youtube.com"});
+    embedded_https_test_server().RegisterRequestHandler(base::BindRepeating(
+        &CreatorDetectionBrowserTest::HandleRequest, base::Unretained(this)));
+    ASSERT_TRUE(embedded_https_test_server().Start());
+  }
+
+  void TearDownOnMainThread() override {
+    EXPECT_TRUE(embedded_https_test_server().ShutdownAndWaitUntilComplete());
+    InProcessBrowserTest::TearDownOnMainThread();
+  }
+
+  using RequestCallback =
+      base::RepeatingCallback<void(std::string_view, BasicHttpResponse&)>;
+
+  template <typename F>
+  void SetRequestCallback(F callback) {
+    request_callback_ = base::BindLambdaForTesting(callback);
+  }
+
+  void SetResponseHTML(std::string_view html) {
+    std::string content(html);
+    SetRequestCallback(
+        [content](std::string_view path, BasicHttpResponse& response) {
+          response.set_content(content);
+        });
+  }
+
+  void SetRewardsEnabled() {
+    auto* prefs = browser()->profile()->GetPrefs();
+    prefs->SetBoolean(brave_rewards::prefs::kEnabled, true);
+  }
+
+  void NavigateTo(std::string_view host, std::string_view path) {
+    auto url = embedded_https_test_server().GetURL(host, path);
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
```

Comment:
This pattern of assertions within helper methods obscures the source of the failure. It's better to return a bool and `ASSERT(NavigateTo(...))`

Reply:
Also, IMHO it's okay to use `CHECK_EQ` if it's related to the correctness of the text fixture (not for the subsystem we're going to verify).
`ASSERT_TRUE(SomeHelperFunction(..))` makes the text harder to read.

Reply:
Fixed.